server/jobs/storage_backend.go

```diff
@@ -20,66 +20,37 @@ const OutputPrefix = "output"
 
 type StorageBackend interface {
 	// Read logs from the storage backend. Must close the reader
-	Read(key string) ([]string, error)
+	Read(ctx context.Context, key string) ([]string, error)
 
 	// Write logs to the storage backend
-	Write(key string, logs []string, fullRepoName string) (bool, error)
+	Write(ctx context.Context, key string, logs []string, fullRepoName string) (bool, error)
 }
 
-func NewStorageBackend(jobs valid.Jobs, logger logging.Logger, featureAllocator feature.Allocator, scope tally.Scope) (StorageBackend, error) {
-
-	if jobs.StorageBackend == nil {
-		return &NoopStorageBackend{}, nil
-	}
-
-	config := jobs.StorageBackend.BackendConfig.GetConfigMap()
-	backend := jobs.StorageBackend.BackendConfig.GetConfiguredBackend()
-	containerName := jobs.StorageBackend.BackendConfig.GetContainerName()
-
-	location, err := stow.Dial(backend, config)
-	if err != nil {
-		return nil, err
-	}
-
-	storageBackend := &storageBackend{
-		location:      location,
-		logger:        logger,
-		containerName: containerName,
-	}
-
+func NewStorageBackend(client *storage.Client, logger logging.Logger, featureAllocator feature.Allocator, scope tally.Scope) (StorageBackend, error) {
 	return &InstrumentedStorageBackend{
-		StorageBackend: storageBackend,
-		scope:          scope.SubScope("storage_backend"),
+		StorageBackend: &storageBackend{
+			client: client,
+			logger: logger,
+		},
+		scope: scope.SubScope("storage_backend"),
 	}, nil
 }
 
 type storageBackend struct {
-	location      stow.Location
-	logger        logging.Logger
-	containerName string
+	client *storage.Client
+	logger logging.Logger
 }
 
-func (s *storageBackend) Read(key string) (logs []string, err error) {
-	s.logger.Info(fmt.Sprintf("reading object for job: %s in container: %s", key, s.containerName))
-
-	container, err := s.location.Container(s.containerName)
+func (s *storageBackend) Read(ctx context.Context, key string) (logs []string, err error) {
```

Comment:
nit: I need to add this to the style guide, but I don't want us to use named return values, this harms readability imo.